### PR TITLE
[FEAT] Add Discard tab to Garbage Collector in Helios

### PR DESCRIPTION
**Pull request by brian-man**
# Description

Periodically players request that something be discardable.  An easy example that comes up periodically is the desire to discards seeds -- especially since the introducing of the daily chest -- so that Kuebiko is more easily withdrawable.

This change introduces the Discard tab to the Garbage Collector.  The initial discardable items are seeds, but it's trivial to add any other items to the discardable list.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Tested with local game using purchase and sell of sunflower, potato, and pumpkin seeds.

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [x] Screenshot if it includes any UI changes
- [x] I have read the contributing guidelines and agree to the T&Cs
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# Screenshots

image

### CODE REVIEW

**Comment by ShiningMinions** [comment]
Solves https://github.com/sunflower-land/sunflower-land/discussions/2486 , thank you!

File: src/features/game/types/garbage.ts
Comment (spencerdezartsmith) [inline src/features/game/types/garbage.ts:64]: Are these new types necessary? You could potentially use the `isSeed` type guard in your action instead. Would prefer to not have another type that duplicates items.

```
  const isSeed = (selected: InventoryItemName): selected is SeedName =>
    isFruitSeed(selected) || selected in CROP_SEEDS();
```

File: src/features/helios/components/garbageCollector/components/GarbageDiscard.tsx
Comment (spencerdezartsmith) [inline src/features/helios/components/garbageCollector/components/GarbageDiscard.tsx:29]: Is this default assignment necessary?

File: src/features/helios/components/garbageCollector/components/GarbageDiscard.tsx
Comment (spencerdezartsmith) [inline src/features/helios/components/garbageCollector/components/GarbageDiscard.tsx:64]: Here you could either use the seeds, fruit seeds constants and grab their keys or you could build a `DISCARD` constant by doing the same. It just helps us not have to manually update code in multiple places.

**Review by spencerdezartsmith — CHANGES_REQUESTED** [review]
Nice initiative. Just a couple comments about reusing already established types and constants. 

This will require core dev work on the backend before it can be merged.

File: src/features/game/types/garbage.ts
Comment (brian-man) [reply src/features/game/types/garbage.ts:64]: The intent was/is to have an explicit list -- like for the Sell tab -- of what is discardable.  Further, the intent (as noted in the PR description) is to provide a framework for the easy addition of other items to the discardable list.

File: src/features/helios/components/garbageCollector/components/GarbageDiscard.tsx
Comment (brian-man) [reply src/features/helios/components/garbageCollector/components/GarbageDiscard.tsx:29]: Modeled on what the Sell tab does.  Would you like to remove the ` = 1` from both of them?

File: src/features/helios/components/garbageCollector/components/GarbageDiscard.tsx
Comment (brian-man) [reply src/features/helios/components/garbageCollector/components/GarbageDiscard.tsx:64]: See previous reply regarding dedicated names for discardable items.

As for initially populating DISCARD via the seed constants, I'm not opposed to that.

Comment (spencerdezartsmith) [reply src/features/helios/components/garbageCollector/components/GarbageDiscard.tsx:64]: Yes, I would prefer that. Maintaining multiple lists always leads to more work and things being forgotten 🙏🏼

**Comment by ShiningMinions** [comment]
@brian-man FYI if you need help with this one I'm available to collab.

File: src/features/game/types/garbage.ts
Comment (brian-man) [reply src/features/game/types/garbage.ts:64]: @Romy#7694 
How do I do this part...

```
import { CROP_SEEDS } from "features/game/types/crops";

export type DiscardName = <what goes here>;
```

I need `DiscardName` for the type for `DiscardGarbageAction`s 2nd parameter.

File: src/features/helios/components/garbageCollector/components/GarbageDiscard.tsx
Comment (brian-man) [reply src/features/helios/components/garbageCollector/components/GarbageDiscard.tsx:64]: Updated

**Comment by c-py** [comment]
@brian-man Thanks for your contribution and raising this PR! I regret to say, I don't think we will go with this right now.

The reason is that there is a cost of engineering time to maintain this API action - and without the use cases available just yet doesn't seem to be worth it. I understand for Kuebiko holder's it can be frustrating to burn though seeds to withdraw, but we don't want to maintain features for a small numbers of players in a niche circumstance.

Feel free to ping me on #coders-chat if you would like to discuss further or if there are use cases I am missing.

**Comment by brian-man** [comment]
@c-py Bummer.  People have been requesting this for months for seeds and other things.

**Pull request closed without merging**
